packages/wasm-dpp/test/unit/document/Document.spec.js

```diff
@@ -1,36 +1,50 @@
-const DataContractFactory = require('@dashevo/dpp/lib/dataContract/DataContractFactory');
-
-const generateRandomIdentifier = require('@dashevo/dpp/lib/test/utils/generateRandomIdentifier');
-
+const JsDataContractFactory = require('@dashevo/dpp/lib/dataContract/DataContractFactory');
+const JsIdentifier = require('@dashevo/dpp/lib/identifier/Identifier');
+const JsDocument = require('@dashevo/dpp/lib/document/Document');
 const DocumentCreateTransition = require(
   '@dashevo/dpp/lib/document/stateTransition/DocumentsBatchTransition/documentTransition/DocumentCreateTransition',
 );
-
-const Identifier = require('@dashevo/dpp/lib/identifier/Identifier');
-
 const protocolVersion = require('@dashevo/dpp/lib/version/protocolVersion');
 const createDPPMock = require('@dashevo/dpp/lib/test/mocks/createDPPMock');
+const generateRandomIdentifier = require('@dashevo/dpp/lib/test/utils/generateRandomIdentifier');
+const lodash = require('lodash');
 
-const Document = require('@dashevo/dpp/lib/document/Document');
+const generateRandomIdentifierAsync = require('../../../lib/test/utils/generateRandomIdentifierAsync');
+const { default: loadWasmDpp } = require('../../../dist');
 
-const hash = require('@dashevo/dpp/lib/util/hash');
-const serializer = require('@dashevo/dpp/lib/util/serializer');
+let DataContractFactory;
+let DataContractValidator;
+let Identifier;
+let Document;
 
 describe('Document', () => {
-  let hashMock;
-  let encodeMock;
   let rawDocument;
   let document;
   let dataContract;
+  let documentJs;
+  let dataContractJs;
+  let rawDocumentJs;
+  let rawDocumentWithBuffers;
+
+  beforeEach(async () => {
+    ({
+      Identifier, Document, DataContractFactory, DataContractValidator,
+    } = await loadWasmDpp());
+  });
 
-  beforeEach(function beforeEach() {
+  beforeEach(async () => {
```

Comment:
I'd write `async function()` instead of `async () =>`. It might seem as if those are the same thing, but they are not. The arrow function captures the closure a bit differently - it sets `this` to whatever `this` is in the higher scope, while `function` doesn't capture `this` at all. This might lead to unexpected test failures in this particular case, as mocha manipulates `this` value for its hooks, for example, to set timeouts. Setting timeouts in mocha won't work if hooks are arrow functions

Reply:
changed, thanks for explaining!